File: export_firebase_users.js
const admin = require('firebase-admin');
const fs = require('fs');

// Fetch the service account key JSON file contents
// You must provide the path to your service account key file here
const serviceAccount = require('./service-account.json');

admin.initializeApp({
  credential: admin.credential.cert(serviceAccount)
});

const auth = admin.auth();

async function exportUsers() {
  console.log("Starting user export...");
  const users = [];
  
  try {
    // List batch of users, 1000 at a time.
    let listUsersResult = await auth.listUsers(1000);
    
    while (listUsersResult.users.length > 0) {
      listUsersResult.users.forEach((userRecord) => {
        users.push({
          uid: userRecord.uid,
          email: userRecord.email,
          emailVerified: userRecord.emailVerified,
          displayName: userRecord.displayName,
          disabled: userRecord.disabled,
          lastSignInTime: userRecord.metadata.lastSignInTime,
          creationTime: userRecord.metadata.creationTime,
          passwordHash: userRecord.passwordHash ? userRecord.passwordHash.toString('base64') : undefined,
          passwordSalt: userRecord.passwordSalt ? userRecord.passwordSalt.toString('base64') : undefined,
        });
      });
      console.log(`Fetched ${users.length} users so far...`);
      
      if (listUsersResult.pageToken) {
        // List next batch of users.
        listUsersResult = await auth.listUsers(1000, listUsersResult.pageToken);
      } else {
        break; // No more users
      }
    }
    
    // Save to JSON file
    fs.writeFileSync('firebase_users_export.json', JSON.stringify(users, null, 2), 'utf8');
    console.log(`Successfully exported ${users.length} users to firebase_users_export.json`);
    
  } catch (error) {
    console.error('Error listing users:', error);
  }
}

exportUsers();

File: export_firebase_users.py
import firebase_admin
from firebase_admin import credentials
from firebase_admin import auth
import json

# Fetch the service account key JSON file contents
# You must provide the path to your service account key file here
cred = credentials.Certificate('firebase-service-account.json')
firebase_admin.initialize_app(cred)

def export_users():
    print("Starting user export...")
    users = []
    # Start listing users from the beginning, 1000 at a time.
    page = auth.list_users()
    while page:
        for user in page.users:
            users.append({
                'uid': user.uid,
                'email': user.email,
                'email_verified': user.email_verified,
                'phone_number': user.phone_number,
                'display_name': user.display_name,
                'disabled': user.disabled,
            })
        print(f"Fetched {len(users)} users so far...")
        # Get next batch of users.
        page = page.get_next_page()
    
    with open('firebase_users_export.json', 'w', encoding='utf-8') as f:
        json.dump(users, f, ensure_ascii=False, indent=4)
        print(f"Successfully exported {len(users)} users to firebase_users_export.json")

if __name__ == '__main__':
    export_users()
